pkg/volume/volume.go

```diff
@@ -132,3 +133,22 @@ func RenameDirectory(oldPath, newName string) (string, error) {
 	}
 	return newPath, nil
 }
+
+// Return the wrapper volume spec of specific volume plugin
+func GetWrappedVolumeSpec(volName string, pluginName string) *Spec {
+	// The name of wrapper volume is set to "wrapped_{wrapped_volume_name}"
+	wrapperVolumeName := fmt.Sprintf("wrapped_%v", volName)
+
+	var volumeSpec *Spec
+	switch pluginName {
+	case "kubernetes.io/downward-api", "kubernetes.io/secret":
```

Comment:
sorry, this is problematic - these volume plugins should be completely opaque to the plugin-support logic.  The core code may never spell out or know anything about the details of any plugin.  You have to pretend that VolumeSource is an opaque blob.

The way I thought of this was that wrappers call `NewWrapperBuilder(name string, spec volume.Spec, ...)` and `NewWrapperCleaner(name string, spec volume.Spec)` and those functions are responsible for formatting the wrap-name and inserting it into the struct before passing it along.

Note that they have to take by value instead of by pointer, so they can mutate the struct (or else they have to make a copy internally).